### DISCUSSION TITLE
tenant cache not work with file drive

### DISCUSSION THREAD

**Discussion by MadhushanWijesuriya** [spatie/laravel-multitenancy#322] [Q&A]
Hi,
cache working fine when it store into tenant database, but it doesn't work when use file as cache driver I already using the PrefixCacheTask but doesn't work. What can be the reason

**Comment by BertKooij** [comment]
You did not provide a lot of information about your setup or your problem, for example: What exactly does not seem to work? Is the cache overridden with each tenant? Or is it simply not hitting the cache at all? Providing some code examples will increase the change of finding out what the problem is.

Please also note that the documentation states:

>This task only works for memory based caches, such as APC and Redis.

https://spatie.be/docs/laravel-multitenancy/v2/using-tasks-to-prepare-the-environment/prefixing-cache

So you might need to tweak the configuration in order to get it working with file based caching. That is because the Laravel `FileStore` simply does nothing with the provided prefix (https://github.com/laravel/framework/blob/8.x/src/Illuminate/Cache/FileStore.php#L349). 

You could try altering the cache path inside a Task, but you have to be careful when using the Cache Facade within for example queues: https://spatie.be/docs/laravel-multitenancy/v2/advanced-usage/using-tenant-specific-facades.

Alternatively you might be able to use a memory based cache for your project.

**Reply by masterix21** [reply]
Why not PR laravel to support the prefix for the file name?

**Reply by MadhushanWijesuriya** [reply]
Hi @BertKooij,
Thanks for the reply....My issue is cache overridden with each tenant. I drop here my config files for get more support on this issue...
`'switch_tenant_tasks' => [
        Spatie\Multitenancy\Tasks\PrefixCacheTask::class,
    ],`
I'm using redis as database driver. Package version 2.0
